asApp/src/save_restore.c

```diff
@@ -1800,7 +1800,15 @@ STATIC int write_it(char *filename, struct chlist *plist)
                 print_chmod_error(err);
             }
         }
-        out_fd = fdopen(filedes, "w");
+        if (!(out_fd = fdopen(filedes, "w"))) {
+            printf("save_restore:write_it - unable to fdopen file '%s': %s [%s]\n",
+                filename, strerror(errno), datetime);
+            if (++save_restoreIoErrors > save_restoreRemountThreshold) {
+                save_restoreNFSOK = 0;
+                strNcpy(SR_recentlyStr, "Too many I/O errors", STATUS_STR_LEN);
+            }
+            return (ERROR);
+        }
```

Comment:
Given that there are three blocks of code that are essentially the same, could you consider refactoring them?

Reply:
Thanks for the feedback. I've shoved a goto in there to clean it up a bit.